src/modules/flow/http-server/http-server.c

```diff
@@ -497,4 +497,77 @@ float_process_cb(struct http_data *mdata, const struct sol_flow_packet *packet)
     return sol_flow_packet_get_drange(packet, &mdata->value.d);
 }
 
+/* ---------------------------  static files ----------------------------------------- */
+
+static int
+static_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    struct http_data *mdata = data;
+    struct sol_flow_node_type_http_server_static_options *opts =
+        (struct sol_flow_node_type_http_server_static_options *)options;
+
+    if (!server) {
+        server = sol_http_server_new(HTTP_SERVER_PORT);
+        SOL_NULL_CHECK(server, -1);
```

Comment:
should use some -errno, maybe `-ENOMEM` or something like that.

also, would be nice to have this and the init_count as a function that is common to all node types.